### PR TITLE
AZP: fix return code

### PR DESCRIPTION
**Pull request by avildema**
## Why
Fix some recent CI issues

## How
- Fix return code check: tests could fail but CI still passed
- Fix ucp_client_server error on Bluefield
- Fix ucp_hello_world compat failure on swx-rain machines
- Temporarily disable a failing test with valgrind

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 1]
```diff
@@ -1223,7 +1223,7 @@ UCS_TEST_P(multi_rail_max, max_lanes, "IB_NUM_PATHS?=16", "TM_SW_RNDV=y",
                               get_bytes_sent(receiver().ep(), i);
 
         /* Verify that each lane sent something */
-        ASSERT_GE(bytes_sent, 50000 / ucs::test_time_multiplier());
+        // ASSERT_GE(bytes_sent, 50000 / ucs::test_time_multiplier());
```
Comment (evgeny-leksikov) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:1226]: maybe add a `TEST_MESSAGE << "FIXME: temporary disabled due to ..."`?

Comment (yosefe) [reply test/gtest/ucp/test_ucp_tag_xfer.cc]: @evgeny-leksikov fixed by force push

**Review by evgeny-leksikov — APPROVED** [review]

**Review by evgeny-leksikov — APPROVED** [review]

**Comment by yosefe** [comment]
@evgeny-leksikov can you pls take a look on latest commits?

**Review by evgeny-leksikov — APPROVED** [review]

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov can you pls take a look on latest commits?

does it fixes ucp_client_server on BF? I can't reproduce it locally

**Comment by yosefe** [comment]
> does it fix ucp_client_server on BF? I can't reproduce it locally

Yes, i could reproduce it on BF (and verified that it fixed)
i guess it's a timing issue

**Pull request merged**